### PR TITLE
fix: openapi3.SchemaError message customize (#678)

### PR DESCRIPTION
**Pull request by micronull**
See #678.
You can, of course, try to use i18n, but then we won't be able to substitute the values we need from the schema fields, such as title.

### CODE REVIEW

File: openapi3/schema_validation_settings.go [hunk 1]
```diff
@@ -50,6 +52,11 @@ func DefaultsSet(f func()) SchemaValidationOption {
 	return func(s *schemaValidationSettings) { s.defaultsSet = f }
 }
 
+// SetSchemaErrorCustomMessage allows to override the schema error message.
+func SetSchemaErrorCustomMessage(f func(err *SchemaError) string) SchemaValidationOption {
```
Comment (fenollp) [inline openapi3/schema_validation_settings.go:56]: `SetSchemaErrorMessageCustomizer`

File: openapi3/schema_test.go [hunk 2]
```diff
@@ -1089,6 +1090,33 @@ func TestSchemaErrors(t *testing.T) {
 	}
 }
 
+func TestSchemaError_CustomMessage(t *testing.T) {
```
Comment (fenollp) [inline openapi3/schema_test.go:1093]: This should be an Example.

File: openapi3/schema.go [hunk 3]
```diff
@@ -1687,6 +1714,12 @@ func (err *SchemaError) JSONPointer() []string {
 }
 
 func (err *SchemaError) Error() string {
+	if err.customizeMessageError != nil {
+		if msg := err.customizeMessageError(err); msg != "" {
```
Comment (fenollp) [inline openapi3/schema.go:1718]: The documentation of this option should mention that returning an empty string falls back to the upstream `Error()` impl.

File: openapi3filter/validate_request_test.go [hunk 4]
```diff
@@ -215,3 +216,53 @@ components:
...
+
+	route, pathParams, err := router.FindRoute(req)
+	require.NoError(t, err)
+
+	validationInput := &RequestValidationInput{
+		Request:    req,
+		PathParams: pathParams,
+		Route:      route,
+		Options: &Options{
+			CustomSchemaErrorFunc: func(err *openapi3.SchemaError) string {
```
Comment (fenollp) [inline openapi3filter/validate_request_test.go:260]: Use the function calls to set the options slice instead of the struct.

**Review by fenollp — CHANGES_REQUESTED** [review]
The ask and approach seem valid to me, let's fix some naming / docs and we'll be ready to go!

File: openapi3filter/validate_request_test.go [hunk 5]
```diff
@@ -215,3 +216,53 @@ components:
 		})
 	}
 }
+
+func TestValidateRequest_CustomSchemaErrorMessage(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (micronull) [reply openapi3filter/validate_request_test.go:220]: turned

File: openapi3/schema_test.go (hunk 2 above)
Comment (micronull) [reply openapi3/schema_test.go]: made an example of.

File: openapi3/schema.go (hunk 3 above)
Comment (micronull) [reply openapi3/schema.go:1718]: added

File: openapi3filter/validate_request_test.go (hunk 4 above)
Comment (micronull) [reply openapi3filter/validate_request_test.go]: made by calling the `WithCustomSchemaErrorFunc` function

**Review by fenollp — APPROVED** [review]

**Pull request merged**